### ISSUE TITLE
handshake error.

### ISSUE THREAD

**Issue by gordanyang** [appleboy/drone-ssh#84]
handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain
drone-ssh error:  ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain

**Comment by appleboy** [comment]
post your sample code.

**Comment by dmitry7887** [comment]
same for me
  ssh:
    image: appleboy/drone-ssh
    host: my.aws.host.ip
    username: ubuntu
    key: .drone/key.pem
    port: 22
    script:
      - .drone/deploy.sh

**Comment by Dineshs91** [comment]
The same error occurs for me too. I am using drone secrets to get the key and host.

**Comment by Dineshs91** [comment]
The same configuration works fine in drone 0.8 without any errors.

**Comment by guyskk** [comment]
I meet the same error:

.drone.yml
```yaml
pipeline:
    ssh:
        image: appleboy/drone-ssh
        host: MY_SERVER_IP
        username: dev
        port: 22
        secrets: [plugin_ssh_key]
        script:
            - echo hello
            - echo world
```

and I set `PLUGIN_SSH_KEY` in `http://ci.myserver.com/{repo}/settings/secret/` page，it's content was copyed from `~/.ssh/id_rsa`.

the build logs:
```
======CMD======
echo hello
echo world
======END======
ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain
drone-ssh error:  ssh: handshake failed: ssh: unable to authenticate, attempted methods [none], no supported methods remain
```

drone version: `houseboy/drone:20170512.1`, it's a fork of drone 0.8 which support https://coding.net .

What can I do for finding out the reason and solving the problem?

**Comment by appleboy** [comment]
@guyskk just set `ssh_key` in secret tab of project setting and update the drone config as the following:

```diff
pipeline:
    ssh:
        image: appleboy/drone-ssh
        host: MY_SERVER_IP
        username: dev
        port: 22
-       secrets: [plugin_ssh_key]
+       secrets: [ ssh_key ]
        script:
            - echo hello
            - echo world
```

**Comment by guyskk** [comment]
@appleboy still the same problem.

I read the code in https://github.com/appleboy/drone-ssh/blob/master/main.go#L32 :
```go
cli.StringFlag{
        Name:   "ssh-key",
        Usage:  "private ssh key",
        EnvVar: "PLUGIN_SSH_KEY,PLUGIN_KEY,SSH_KEY",
},
```
`PLUGIN_SSH_KEY,PLUGIN_KEY,SSH_KEY` seems equivalent.

**Comment by appleboy** [comment]
Please make sure that drone server can connect ssh server using the command line.

**Comment by guyskk** [comment]
@appleboy Yes, I can connect via `ssh dev@MY_SERVER_IP -i drone_id_rsa` in drone server, `drone_id_rsa` was copyed from `~/.ssh/id_rsa` in my laptop.

**Comment by appleboy** [comment]
add `pull: true` in your config to make sure drone-ssh is the latest version.

**Comment by guyskk** [comment]
@appleboy still the same.

**Comment by guyskk** [comment]
@appleboy the problem fixed after I change `houseboy/drone:20170512.1` to my manual build latest drone, probably `houseboy/drone:20170512.1` is not fresh enough. 
Thank you for your help!

**Comment by appleboy** [comment]
@gordanyang Any updates on this?
